internal/util/diffmap.go

```diff
@@ -0,0 +1,18 @@
+package util
+
+func ExtractResourceVersion(objectAsMap map[string]any) (string, bool) {
```

Comment:
please add godoc to this exported function

Reply:
done in 0e5217a